tsdb/querier.go

```diff
@@ -247,7 +252,7 @@ func PostingsForMatchers(ctx context.Context, ix IndexReader, ms ...*labels.Matc
 			return nil, ctx.Err()
 		}
 		switch {
-		case m.Name == "" && m.Value == "": // Special-case for AllPostings, used in tests at least.
+		case m.Name == "" && m.Value == "":
```

Comment:
I meant take this case out completely, not just the comment.

Reply:
Oh, I see.
I don't know if we can simply remove this case as it fallbacks to the default case.
Is it better if I keep this case but do nothing?

Reply:
Could raise an error; we don't think this can happen in real PromQL (aside from the case handled at the top).

Reply:
Updated